### PR TITLE
Rename many global enums relating to input

### LINKED ISSUE #1817: Rename many enums relating to input, especially Button to MouseButton
**Describe the project you are working on:**

This affects any project using these input enums.

**Describe the problem or limitation you are having in your project:**

The names of these enums are somewhat inconsistent/confusing and can use some improvements.

With the input enums, we currently have `ButtonList`, `KeyList`, `JoyButtonList`, `JoyAxisList`, and `MidiMessageList`. However, most enums in Godot don't have the `List` suffix. As seen [here](https://docs.godotengine.org/en/stable/classes/class_@globalscope.html), `Margin`, `Corner`, `Orientation`, `HAlign`, `VAlign`, `KeyModifierMask`, `Error`, `PropertyHint`, `PropertyUsageFlags`, `MethodFlags`, `Variant.Type`, and `Variant.Operator` do not contain the `List` suffix. Additionally, other enums not listed there like `Vector3::Axis` do not contain the `List` suffix.

The biggest issue is specifically with `ButtonList` which has members starting with `BUTTON_`. Just from the name, it's hard to tell what this actually is. Keyboard buttons? Controller buttons? UI buttons on your screen? It actually refers to mouse buttons, like left click and right click. The only way you would know that this is referring to mouse buttons is if you're already familiar with Godot.

**Describe the feature / enhancement and how it helps to overcome the problem or limitation:**

If we remove the `List` suffix from the input enums, they will be consistent with the other enums.

If we renamed `ButtonList` to `MouseButton` and have the members start with `MOUSE_BUTTON_`, then it will be more clear that this enum is for mouse buttons.

Since this is a rename, this breaks compatibility, so it should be decided on before Godot 4.0 comes out.

**Describe how your proposal will work, with code, pseudocode, mockups, and/or diagrams:**

Branch: https://github.com/aaronfranke/godot/tree/enums

PR: https://github.com/godotengine/godot/pull/38054

**If this enhancement will not be used often, can it be worked around with a few lines of script?:**

These are used often. The workaround is to use the current names.

**Is there a reason why this should be core and not an add-on in the asset library?:**

These are core enums, so they are already in core.

### PR DESCRIPTION
**Pull request by aaronfranke**
Implements https://github.com/godotengine/godot/issues/16863#issuecomment-612792875 + the following 2 comments. Implements and closes https://github.com/godotengine/godot-proposals/issues/1817.

* `ButtonList` -> `MouseButton` and members now start with `MOUSE_BUTTON_`

* `KeyList` -> `Key` and all members are the same.

* `JoyButtonList` -> `JoyButton` and `JoyAxisList` -> `JoyAxis`.

* `MidiMessageList` -> `MIDIMessage` for consistency with the above.

### CODE REVIEW

**Comment by Calinou** [comment]
Shouldn't `MidiMessage` be `MIDIMessage`? We already have `HTTPRequest`, `XMLParser`, and so on.

File: core/input/input.cpp [hunk 1]
```diff
@@ -1206,22 +1206,22 @@ void Input::_get_mapped_hat_events(const JoyDeviceMapping &mapping, int p_hat, J
...
-			return JoyButtonList(i);
+			return JoyButton(i);
 		}
 	}
-	return JoyButtonList::JOY_BUTTON_INVALID;
+	return JoyButton::JOY_BUTTON_INVALID;
 }
 
-JoyAxisList Input::_get_output_axis(String output) {
+::JoyAxis Input::_get_output_axis(String output) {
```
Comment (akien-mga) [inline core/input/input.cpp:1218]: It's a bit unfortunate to have a name clash with the other `JoyAxis`. Not sure if/how to solve it, it's not too bad I guess.

**Review by akien-mga — APPROVED** [review]
Seems like a good change to me overall. Would like some extra opinions before merging though.

File: core/input/input.cpp (hunk 1 above)
Comment (aaronfranke) [reply core/input/input.cpp:1218]: We could rename `struct JoyAxis` to something else.

Comment (akien-mga) [reply core/input/input.cpp:1218]: Good idea, maybe `JoyAxisValue`?

File: core/core_constants.cpp [hunk 2]
```diff
@@ -104,12 +104,12 @@ static Vector<_CoreConstant> _global_constants;
 
 #endif
 
-VARIANT_ENUM_CAST(KeyList);
+VARIANT_ENUM_CAST(Key);
```
Comment (akien-mga) [inline core/core_constants.cpp:107]: Only remaining concern that I have is that some users might not be happy with the fairly generic (3 letter) `Key` identifier being taken in the global scope. Might lead to issues if one wants to name a class `Key` for a key item, or naming a public property `Key` in C# (which I think uses PascalCase for public properties).

Comment (aaronfranke) [reply core/core_constants.cpp:107]: We also already have name conflicts between properties and non-properties [in the core Godot API for C#](https://github.com/godotengine/godot-proposals/issues/1814).

One option is to make `KeyList` be the exception, or we could do something else like `Keyboard` or `KeyCode`. `Key` seems fine to me though.

Comment (KoBeWi) [reply core/core_constants.cpp:107]: `KeyCode` sounds good. InputEventKey has a property with this name.

Comment (akien-mga) [reply core/core_constants.cpp:107]: Yeah I guess `KeyCode` might be fine. It does re-add an inconsistency, but to some extent the `KeyList` enumerated values are actually different from the other enums. They're internal keycodes while the rest are indices (0 to max).

Comment (aaronfranke) [reply core/core_constants.cpp:107]: Should we have `KEY_ENTER` or `KEY_CODE_ENTER`?

Comment (KoBeWi) [reply core/core_constants.cpp:107]: `KEY_ENTER`

Comment (aaronfranke) [reply core/core_constants.cpp:107]: Oh, we can't use `KeyCode`, it conflicts with X11:

```
/usr/include/X11/X.h:108:23: note: 'KeyCode' has a previous declaration here
  108 | typedef unsigned char KeyCode;
      |                       ^~~~~~~
```

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**